src/mlpack/core/math/digamma.hpp

```diff
@@ -0,0 +1,256 @@
+ * @author Gopi Tatiraju 
+ *
+ * Some parts of the implementation are inspired from boost.
+ * Boost Software License, Version 1.0. (See accompanying file
+ * LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+ * 
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_MATH_DIGAMMA_HPP
+#define MLPACK_CORE_MATH_DIGAMMA_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace math {
+
+
+/**
+ * This function evaluates the polynomial based on the
+ * constants passed to the function.
+ * This function evaluates the polynomial when x > 10.
+ *
+ * @param a Array of constants.
+ * @param x Input for which digamma will be calculated.
+ */
+template<std::size_t N, typename T>
+typename std::enable_if<N == 8, T>::type
+EvaluatePolyLarge(const T(&a)[N], const T& x)
+{
+  T x2 = x * x;
+  T t[2];
+  
+  t[0] = a[7] * x2 + a[5];
+  t[0] *= x2;
+  t[0] += a[3];
+  t[0] *= x2;
+  t[0] += a[1];
+  t[0] *= x;
+  
+  t[1] = a[6] * x2 + a[4];
+  t[1] *= x2;
+  t[1] += a[2];
+  t[1] *= x2;
+  t[1] += a[0];
+  
+  return t[0] + t[1];
+}
+
+/**
+ * This function evaluates the polynomial based on the
+ * constants passed to the function.
+ * This gets executed when size of the array of constants is 7.
+ * This function evaluates the polynomial
+ * when x is in the interval [1, 2].
+ *
+ * @param a Array of constants.
+ * @param x Input for which digamma will be calculated.
+ */
+template<std::size_t N, typename T>
+typename std::enable_if<N == 7, T>::type
+EvaluatePoly_1_2(const T(&a)[N], const T& x)
+{
+  T x2 = x * x;
+  T t[2];
+
+  t[0] = (a[6] * x2) + a[4];
+  t[0] *= x2;
+  t[0] += a[2];
+  t[0] *= x2;
+  t[0] += a[0];
+
+  t[1] = (a[5] * x2) + a[3];
+  t[1] *= x2;
+  t[1] += a[1];
+  t[1] *= x;
+
+  return t[0] + t[1];
+}
+
+/**
+ * This function evaluates the polynomial based on the
+ * constants passed to the fucntion.
+ * This gets executed when size of the array of constants is 6.
+ * This function is to evaluate the polynomial when x
+ * is in the interval [1, 2].
+ *
+ * @param a Array of constants.
+ * @param x Input for which digamma will be calculated.
+ */
+template<std::size_t N, typename T>
+typename std::enable_if<N == 6, T>::type
+EvaluatePoly_1_2(const T(&a)[N], const T& x)
+{
+  T x2 = x * x;
+  T t[2];
+
+  t[0] = (a[5] * x2) + a[3];
+  t[0] *= x2;
+  t[0] += a[1];
+  t[0] *= x;
+
+  t[1] = (a[4] * x2) + a[2];
+  t[1] *= x2;
+  t[1] += a[0];
+
+  return t[0] + t[1];
+}
+
+/**
+ * This function calculates and returns Digamma(x)
+ * when x is in the interval [1, 2].
+ *
+ * @param x Input for which digamma will be calculated.
+ */
+template<typename T>
+T Digamma_1_2(T x)
+{
+  T y = 0.99558162689208984F;
+
+  const T root_1 = T(1569415565) / 1073741824uL;
```

Comment:
I know I am being *super* picky, but do you think you can change the names to conform with the mlpack style?  (I feel a little bad asking because I know it's tedious, but I do think consistency is useful and important.  If you like, when I have time I'm happy to pass through and try to do it as suggestions.)